api/src/main/java/ca/bc/gov/educ/api/graduation/model/report/ReportData.java

```diff
@@ -1,96 +1,81 @@
-    private String gradMessage;
-    private String updateDate;
-    private Certificate certificate;
-
-    public Student getStudent() {
-        return student;
-    }
-
-    public void setStudent(Student value) {
-        this.student = value;
-    }
-
-    public School getSchool() {
-        return school;
-    }
-
-    public void setSchool(School value) {
-        this.school = value;
-    }
-
-    public String getLogo() {
-        return logo;
-    }
-
-    public void setLogo(String value) {
-        this.logo = value;
-    }
-
-    public List<NonGradReason> getNonGradReasons() {
-        return nonGradReasons;
-    }
-
-    public void setNonGradReasons(List<NonGradReason> value) {
-        this.nonGradReasons = value;
-    }
-
-    public Transcript getTranscript() {
-        return transcript;
-    }
-
-    public void setTranscript(Transcript value) {
-        this.transcript = value;
-    }
-
-    public GradProgram getGradProgram() {
-        return gradProgram;
-    }
-
-    public void setGradProgram(GradProgram value) {
-        this.gradProgram = value;
-    }
-
-    public GraduationData getGraduationData() {
-        return graduationData;
-    }
-
-    public void setGraduationData(GraduationData value) {
-        this.graduationData = value;
-    }
-
-    public String getGradMessage() {
-        return gradMessage;
-    }
-
-    public void setGradMessage(String value) {
-        this.gradMessage = value;
-    }
-
-    public String getUpdateDate() {
-        return updateDate;
-    }
-
-    public void setUpdateDate(String value) {
-        this.updateDate = value;
-    }
-
-    public Certificate getCertificate() {
-        return certificate;
-    }
-
-    public void setCertificate(Certificate value) {
-        this.certificate = value;
-    }
+import java.util.Map;
+
+@Data
+@Component
+@XmlType(name = "")
+@XmlRootElement(name = "generateReport")
+@XmlSeeAlso({
+		Student.class,
+		School.class,
+		Transcript.class,
+		GradProgram.class,
+		NonGradReason.class,
+		Certificate.class
+})
+@JsonSerialize(include=JsonSerialize.Inclusion.NON_NULL)
+//@JsonPropertyOrder(alphabetic = true)
+//@JsonRootName("generateReport")
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "type")
+@JsonSubTypes({
+		@JsonSubTypes.Type(value = Student.class),
+		@JsonSubTypes.Type(value = School.class),
+		@JsonSubTypes.Type(value = Transcript.class),
+		@JsonSubTypes.Type(value = GradProgram.class),
+		@JsonSubTypes.Type(value = NonGradReason.class),
+		@JsonSubTypes.Type(value = Certificate.class),
+		@JsonSubTypes.Type(value = OptionalProgram.class),
+		@JsonSubTypes.Type(value = GradRequirement.class),
+		@JsonSubTypes.Type(value = Exam.class),
+		@JsonSubTypes.Type(value = Assessment.class),
+		@JsonSubTypes.Type(value = AssessmentResult.class),
+		@JsonSubTypes.Type(value = AchievementCourse.class)
+})
+public class ReportData implements Serializable {
+
```

Comment:
Hi Sreepad, can you implement serializable on all properties? For example, School, Transcript, GradProgram, etc should also implement serializable and have a 'private static final long serialVersionUID = {long}L' property for serialization compliance.

Reply:
I have made the changes